fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -178,54 +178,110 @@ class BrowserRobot {
+            )
+            // Open link in new tab option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_open_link_in_new_tab))),
+                waitingTime,
+            )
+            // Open link in private tab option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_open_link_in_private_tab))),
+                waitingTime,
+            )
+            // Copy link option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_copy_link))),
+                waitingTime,
+            )
+            // Download link option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_download_link))),
+                waitingTime,
+            )
+            // Share link option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_share_link))),
+                waitingTime,
+            )
+            // If the link is not re-directing to another local asset the "Download link" option is available
+            // If the link is re-directing to an external app the "Open link in external app" option is available
         } else {
+            // Link URL
             mDevice.waitNotNull(
                 Until.findObject(By.textContains(containsURL.toString())),
                 waitingTime,
             )
+            // Open link in new tab option
             mDevice.waitNotNull(
                 Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_open_link_in_new_tab))),
                 waitingTime,
             )
+            // Open link in private tab option
             mDevice.waitNotNull(
                 Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_open_link_in_private_tab))),
                 waitingTime,
             )
+            // Copy link option
             mDevice.waitNotNull(
                 Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_copy_link))),
                 waitingTime,
             )
+            // Download link option
             mDevice.waitNotNull(
                 Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_download_link))),
                 waitingTime,
             )
+            // Share link option
             mDevice.waitNotNull(
                 Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_share_link))),
                 waitingTime,
             )
+            // Open in external app option
+            mDevice.waitNotNull(
+                Until.findObject(text(getStringResource(R.string.mozac_feature_contextmenu_open_link_in_external_app))),
+                waitingTime,
+            )
```

Comment:
Is this a single function on 100 lines? I think we should see a way of splitting this into smaller, individual ones.